libcontainer/cgroups/fs2/memory.go

```diff
@@ -104,10 +104,17 @@ func statMemory(dirPath string, stats *cgroups.Stats) error {
 
 	memoryUsage, err := getMemoryDataV2(dirPath, "")
 	if err != nil {
-		if errors.Is(err, unix.ENOENT) && dirPath == UnifiedMountpoint {
+		if errors.Is(err, unix.ENOENT) && dirPath == rootCgroupPath {
 			// The root cgroup does not have memory.{current,max}
 			// so emulate those using data from /proc/meminfo.
-			return statsFromMeminfo(stats)
+			if err := statsFromMeminfo(stats); err != nil {
+				return err
```

Comment:
I don't like that the logic of getting memory stats for root cgroup is now spread between `statsFromMeminfo()` and this function. Can you please move the logic you add below into `statsFromMeminfo` instead of making an addition here?

Reply:
yep!